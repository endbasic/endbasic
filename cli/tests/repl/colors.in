' EndBASIC
' Copyright 2020 Julio Merino
'
' Licensed under the Apache License, Version 2.0 (the "License"); you may not
' use this file except in compliance with the License.  You may obtain a copy
' of the License at:
'
'     http://www.apache.org/licenses/LICENSE-2.0
'
' Unless required by applicable law or agreed to in writing, software
' distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
' WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
' License for the specific language governing permissions and limitations
' under the License.

' Tests that the background color is used to clear all new lines until the right
' margin of the screen.

A = 5: WHILE A > 0: COLOR A, A - 1: PRINT "Hello": A = A - 1: END WHILE
COLOR

COLOR 4, 2
INPUT "Number? "; num
123

PRINT "Message"
COLOR
