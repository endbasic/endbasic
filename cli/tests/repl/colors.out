
    Welcome to EndBASIC X.Y.Z.

    Type HELP for interactive usage information.
    For a guided tour, type: LOAD "DEMOS:/TOUR.BAS": RUN

[38;5;5m[48;5;4mHello[K
[38;5;4m[48;5;3mHello[K
[38;5;3m[48;5;2mHello[K
[38;5;2m[48;5;1mHello[K
[38;5;1m[48;5;0mHello[K
[39m[49m[38;5;4m[48;5;2mMessage[K
[39m[49mEnd of input by CTRL-D
